### PR TITLE
*: add install script.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay @disksing @zhangjinpeng1987

### CODE REVIEW

File: USAGE.md [hunk 1]
```diff
@@ -78,6 +80,9 @@
     ```
 
     The binary of `tidb-server` is installed in `${GOPATH}/src/github.com/pingcap/tidb/tidb-server`
+    
+You can also use [install.sh](./install.sh) to install above components. The binary
```
Comment (ngaut) [inline USAGE.md]: Move up, please

**Comment by ngaut** [comment]
LGTM

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
```
Comment (ngaut) [inline install.sh]: Makefile is just for TiKV.

Comment (BusyJay) [inline install.sh]: Then why the installation instructions of TiDB or Rocksdb are included in TiKV project?

Comment (ngaut) [inline install.sh]: User needs to use them together. The only driver of TiKV is written in go.

File: .gitignore [hunk 3]
```diff
@@ -22,3 +22,5 @@ out/
 .vscode/**
 
 target
+tmp
```
Comment (BusyJay) [inline .gitignore]: Can we use **dist** or **target** directory instead?

File: scripts/build_rocksdb.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+echo "building rocksdb..."
+cd $DEPS_PATH
+mkdir -p ${DEPS_PATH}
+
+case "$OSTYPE" in 
+    linux*) 
+        if [ ! -d rocksdb-rocksdb-4.6.1 ]; then
+            sudo apt-get update
```
Comment (BusyJay) [inline scripts/build_rocksdb.sh]: Is it appropriate to use sudo in build/install scripts?

**Comment by siddontang** [comment]
PTAL

**Comment by BusyJay** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

File: USAGE.md [hunk 5]
```diff
@@ -9,75 +9,8 @@
...
-+ **TiDB**
-
-    ```sh
-    git clone https://github.com/pingcap/tidb.git ${GOPATH}/src/github.com/pingcap/tidb
-    cd ${GOPATH}/src/github.com/pingcap/tidb
-    make server
-    ```
-
-    The binary of `tidb-server` is installed in `${GOPATH}/src/github.com/pingcap/tidb/tidb-server`
+You can use `make install` to install all binary executions in `bin` directory.
```
Comment (queenypingcap) [inline USAGE.md]: +You can use the `make install` command to install all binary executions in the `bin` directory.

**Pull request merged**